rustworkx-core/src/generators/simple_shortest_paths.rs

```diff
@@ -0,0 +1,368 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use crate::petgraph::algo::{Measure};
+use crate::petgraph::graph::{DiGraph, NodeIndex};
+use crate::petgraph::visit::{EdgeRef, IntoEdges, VisitMap, Visitable};
+
+use std::cmp::Ordering;
+use std::collections::hash_map::Entry::{Occupied, Vacant};
+use std::collections::{BinaryHeap, HashMap};
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::{f32, thread};
+// use rand::Rng;
+///////////
+//             SimplePath
+// This is Structure which saves all the context about graph & iterating attributes
+// Score : the total weightage of the current shortest path
+// Path: Shorted path caulculated in this iteration
+// index: used for iterating over edged to delete one and calculate path once again
+// Source: to store the start point
+// Target: to store goal of path
+// graph: store the path to be used
+// unique_path: stores all unique_paths to verify that we are not returning same path again after random edge removal
+// switch : used for switching to next shortest path in case for one all possible paths are generated.
+/////////////
+
+#[derive(Debug, Clone)]
+pub struct SimplePath {
```

Comment:
Typically the expectation of this kind of functionality would be to have a a public function to serve as the main interface for this functionality that takes in a reference to a graph object and returns an iterator structure like this.

Reply:
I will try that, as yield is not there , I am still looking how to do this, i want to calculate only 1 path at a time, want to go for next path only if .next() or something similar is called.